gh-badges/lib/make-badge.js

```diff
@@ -108,75 +25,47 @@ module.exports = function makeBadge({
   // String coercion and whitespace removal.
   text = text.map(value => `${value}`.trim())
 
-  let [left, right] = text
+  const [label, message] = text
 
-  color = normalizeColor(color)
-  labelColor = normalizeColor(labelColor)
+  color = normalizeColor(color || colorB || colorscheme)
+  labelColor = normalizeColor(labelColor || colorA)
 
   // This ought to be the responsibility of the server, not `makeBadge`.
   if (format === 'json') {
     return JSON.stringify({
-      label: left,
-      message: right,
+      label,
+      message,
       logoWidth,
       color,
       labelColor,
       link: links,
-      name: left,
-      value: right,
+      name: label,
+      value: message,
     })
   }
 
-  if (!(template in templates)) {
-    if (template === 'popout-square') {
-      template = 'flat-square'
-    } else {
-      template = 'flat'
-    }
-  }
-  if (template === 'social') {
-    left = capitalize(left)
-  } else if (template === 'for-the-badge') {
-    left = left.toUpperCase()
-    right = right.toUpperCase()
+  template = camelcase(template)
+  if (template.startsWith('popout')) {
```

Comment:
Yeah good question.. I hadn't really questioned the behaviour of this for library users. As implemented, this will do the same thing for library users and shields.io users. i.e: as implemented a call to `makeBadge({text: ['build', 'passed'], color: 'green', template: 'popout-square' })` will render a `flat-square` badge, but we could break it and force lib users to make that choice as part of their upgrade instead of making an implicit assumption. Its more explicit/less "magic" but also maybe less convenient. I suppose given this realase changes the public interface by replacing `BadgeFactory.create()` with `makeBadge()`, you're going to have to change all your code anyway..

Would that be your preference?

Reply:
Yea. I doubt library users are using popout to begin with, though if they are I think it's a better idea to fail loudly than silently give them what they aren't expecting.

Reply:
93115dd